pkg/server/skydb/recordutil/recordutil.go

```diff
@@ -328,21 +328,33 @@ func RecordSaveHandler(req *RecordModifyRequest, resp *RecordModifyResponse) sky
 	makeAssetsCompleteAndInjectSigner(db, req.Conn, records, req.AssetStore)
 
 	// execute after save hooks
-	if req.HookRegistry != nil {
-		records = executeRecordFunc(records, resp.ErrMap, func(record *skydb.Record) (err skyerr.Error) {
-			originalRecord, _ := originalRecordMap[record.ID]
-			err = req.HookRegistry.ExecuteHooks(req.Context, hook.AfterSave, record, originalRecord)
-			if err != nil {
+	records = hookTriggerer.triggerIfRegistered(records, hook.AfterSave, true)
+
+	resp.SavedRecords = records
+	resp.SchemaUpdated = schemaExtended
+
+	return nil
+}
+
+type saveHookTriggerer struct {
+	Req               *RecordModifyRequest
+	Resp              *RecordModifyResponse
+	OriginalRecordMap map[skydb.RecordID]*skydb.Record
+}
+
+func (t saveHookTriggerer) triggerIfRegistered(records []*skydb.Record, kind hook.Kind, shouldLogErr bool) []*skydb.Record {
```

Comment:
1. If `{A}` and `{B}` does not imply one another, I would agree. However, in this case {A} refers to trigger hooks, and {B} refers to (if) hook is registered. If a hook is not registered, it doesn’t exist, and there is nothing to trigger.
2. The function is named `triggerIfRegistered`, but the code says `if t.Req.HookRegistry != nil`. Doesn’t look like name is a description of the logic. (`triggerIfHookRegistryExists`? No just kidding.) 

So I am not saying that `DoIfSomething` is always a bad practice, no. I am just saying that in this particular case, `triggerIfRegistered` is two words more than necessary. Anyway this is not something I would force on you.